spec/lib/extensions/ar_migration_spec.rb

```diff
@@ -1,109 +1,93 @@
-      end
-
-      it "doesn't blow up when there is no region" do
-        MiqRegion.destroy_all
-        MiqRegion.my_region_clear_cache
-        described_class.update_local_migrations_ran(version, :up)
-      end
-    end
-  end
 
-  describe ArPglogicalMigration::RemoteRegionMigrationWatcher do
-    include_context "with a dummy version"
+      context "with the schema_migrations_ran table" do
+        include_context "with a dummy version"
 
-    let(:subscription) { double("Subscription", :enable => nil, :disable => nil, :provider_region => my_region.region) }
+        it "adds the given version when the direction is :up" do
+          described_class.update_local_migrations_ran(version, :up)
+          expect(described_class.discover_schema_migrations_ran_class.where(:version => version).exists?).to eq(true)
+        end
 
-    subject do
-      described_class.new(subscription, version).tap do |s|
-        allow(s).to receive_messages(:puts => nil, :print => nil)
+        it "doesn't blow up when there is no region" do
+          MiqRegion.destroy_all
+          MiqRegion.my_region_clear_cache
+          described_class.update_local_migrations_ran(version, :up)
+        end
       end
     end
 
-    describe "#wait_for_remote_region_migrations" do
-      context "without the migrations ran column present" do
-        include_context "without the migrations ran column"
+    describe ArPglogicalMigrationHelper::RemoteRegionMigrationWatcher do
+      include_context "with a dummy version"
+      let(:other_region_number) { ArPglogicalMigrationHelper.discover_schema_migrations_ran_class.my_region_number + rand(1..50) }
+      let(:subscription) { double("Subscription", :enable => nil, :disable => nil, :provider_region => other_region_number) }
 
-        it "does nothing" do
-          expect(Vmdb.rails_logger).not_to receive(:info)
-          subject.wait_for_remote_region_migration
+      subject do
+        described_class.new(subscription, version).tap do |s|
+          allow(s).to receive_messages(:puts => nil, :print => nil)
         end
       end
 
-      it "sleeps until the migration is added" do
-        allow(subject).to receive(:restart_subscription)
-        allow(subject.region).to receive(:reload)
+      describe "#wait_for_remote_region_migration" do
+        def wait_for_migration_called
+          @count ||= 0
+          if @count == 5
+            other_region_id = other_region_number * ApplicationRecord.rails_sequence_factor + 1
+            ArPglogicalMigrationHelper.discover_schema_migrations_ran_class.create!(:id => other_region_id, :version => version)
+          end
+          @count += 1
+        end
 
-        subject.region.update_attributes!(:migrations_ran => nil)
+        context "without the schema_migrations_ran table present" do
+          include_context "without the schema_migrations_ran table"
+          let(:other_region_number) { 99 }
```

Comment:
Instead of defining this, could we use `ApplicationRecord.my_region_number + 1` in the `wait_for_migration_called` method?